testsuite/drivers/helpers.py

```diff
@@ -293,3 +294,80 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         import fcntl
         fcntl.flock(self.lock_file.fileno(), fcntl.LOCK_UN)
         self.lock_file.close()
+
+
+GIT_WRAPPER_TEMPLATE = """\
+#! /usr/bin/env python
+import subprocess, sys
+substitution_dict = {substitution_dict}
+# Argument substitutions
+args = sys.argv[1:]
+for key in substitution_dict:
+    args = [arg.replace(key, substitution_dict[key]) for arg in args]
+# Run git
+p = subprocess.run(['{actual_git_path}'] + args, capture_output=True)
+# Output substitutions
+stdout, stderr = p.stdout.decode(), p.stderr.decode()
+for key in substitution_dict:
+    stdout = stdout.replace(substitution_dict[key], key)
+    stderr = stderr.replace(substitution_dict[key], key)
+print(stdout, end="")
+print(stderr, file=sys.stderr, end="")
+# Exit with appropriate error code
+sys.exit(p.returncode)
+"""
+
+class MockGit:
+    """
+    NON-WINDOWS-ONLY
+    A context manager which mocks the git command with string substitutions.
+
+    The string substitutions are specified by the dictionary substitution_dict.
+    Every non-overlapping occurrence of each of its keys in a command line
+    argument is replaced with its corresponding value before being passed to
+    git. The reverse substitution is applied to git's output. The substitutions
+    are applied in the order in which they appear in substitution_dict.
+
+    The mocked version of git will be placed in mock_git_dir, which will be
+    temporarily added to PATH.
+    """
+
+    def __init__(self, substitution_dict, mock_git_dir):
+        self._substitution_dict = substitution_dict
+        self._mock_git_dir = mock_git_dir
+
+    def __enter__(self):
+        # Mocking on Windows would require git.exe wrapper
```

Comment:
In Windows, non-executable scripts cannot in general be called directly, so placing a `git`, `git.py`, `git.bat` or similar script on `PATH` does not have the desired effect (with or without modifications to `PATHEXT`).

To be clear, there's no reason it can't be done, but it seemed more trouble than it's worth for a test that ought not be particularly platform dependent. If you want, I can look into `gprbuild`ing a binary wrapper instead of writing it to a python script.

Reply:
Alright, I see. 

> a test that ought not be particularly platform dependent

I keep being surprised by how quirky Windows can be in the most unexpected ways, but let's not complicate this test any further. If evidence arises of issues on Windows we can do it then.